OCPBUGS#29279: Doc improvements (#71446)
openshift/openshift-docs: modules/lvms-restoring-volume-snapshots.adoc

// Module included in the following assemblies:
//
// storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc

:_mod-docs-content-type: PROCEDURE
[id="lvms-restoring-volume-snapshots_{context}"]
= Restoring volume snapshots

> You can restore a volume snapshot to a persistent volume claim (PVC). When you restore a volume snapshot, {lvms} creates a new PVC.

Reviewer: ```suggestion
You can restore a volume snapshot by creating a new persistent volume claim (PVC) resource having dataSource set as volume snapshot name.
```

Author: Updated as per the suggestion.

The restored PVC is independent of the volume snapshot and the source PVC.

.Prerequisites

* You have created a volume snapshot.

.Procedure

. Log in to the OpenShift CLI (`oc`).
+
[IMPORTANT]
====
If you are using {lvms} in {rh-rhacm}, log in to the {rh-rhacm} CLI using your {product-title} credentials.
====

. Create a `PersistentVolumeClaim` custom resource (CR) YAML file with the configuration to restore the volume snapshot to a PVC:
+
.Example `PersistentVolumeClaim` CR to restore a volume snapshot to a PVC.
[source,yaml]
----
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: lvm-block-1-restore
spec:
  accessModes:
  - ReadWriteOnce
  volumeMode: Block
  Resources:
    Requests:
      storage: 2Gi <1>
  storageClassName: lvms-vg1 <2>
  dataSource:
    name: lvm-block-1-snap <3>
    kind: VolumeSnapshot
    apiGroup: snapshot.storage.k8s.io
----

> <1> Specify the storage of the new PVC that is created upon restoring the volume snapshot. The size of the requested PVC must be the same as that of the volume snapshot that you want to restore. If a larger PVC is required, you can resize the PVC after restoring the volume snapshot.

Author: Updated as per the suggestion.

> <2> Set this field to the value of the `volumeSnapshotClassName` field in the `VolumeSnapshotClass` CR of the volume snapshot that you want to restore.

Reviewer: ```suggestion
<2> Set this field to the value of the `storageClassName` field in the source `PVC` CR of the volume snapshot that you want to restore.
```

> <3> Set this filed to the name of the volume snapshot that you want to restore. 

Reviewer: ```suggestion
<3> Set this field to the name of the volume snapshot that you want to restore. 
```

Author: Updated as per the suggestion.

. Create the `PersistentVolumeClaim` CR in the namespace where you created the the volume snapshot by running the following command:
+
[source,terminal]
----

> # oc create -f <file_name> <namespace>

Reviewer: ```suggestion
# oc create -f <file_name> -n <namespace>
```

Author: Updated as per the suggestion.

// Module included in the following assemblies:
//
// storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc

:_mod-docs-content-type: PROCEDURE
[id="lvms-restoring-volume-snapshots_{context}"]
= Restoring volume snapshots

> To restore a volume snapshot, you must create a `PersistentVolumeClaim` custom resource (CR) with the `dataSource.name` field set to the name of the volume snapshot. 

Reviewer 2: ```suggestion
To restore a volume snapshot, you must create a `PersistentVolumeClaim` with the `dataSource.name` field set to the name of the volume snapshot. 
```
`PersistentVolume` and `PersistentVolumeClaim` are native objects so no custom resources :)

Author: Updated as per the suggestion.

> When you restore a volume snapshot, {lvms} creates a new persistent volume claim (PVC). The restored PVC is independent of the volume snapshot and the source PVC.

Reviewer 2: LVMS does not create another PVC. The restored PVC is the one that the user just created.

Author: Removed the following line:
"When you restore a volume snapshot, {lvms} creates a new persistent volume claim (PVC)."
